### PR TITLE
[homekit] Fix temperature unit conversion for min/max/step

### PR DESCRIPTION
**Pull request by ccutrer**
use the item's unit if it has one. or allow explicit units

### CODE REVIEW

**Review by yfre — APPROVED** [review]
thank you. probably at same point we should remove binding specific setting for temperature unit and rely on item or system settings 
LGTM

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java [hunk 1]
```diff
@@ -604,8 +609,13 @@ private static Supplier<CompletableFuture<Double>> getTemperatureSupplier(Homeki
 
     private static ExceptionalConsumer<Double> setTemperatureConsumer(HomekitTaggedItem taggedItem) {
         return (value) -> {
-            if (taggedItem.getBaseItem() instanceof NumberItem) {
-                taggedItem.send(new DecimalType(convertFromCelsius(value)));
+            Item baseItem = taggedItem.getBaseItem();
+            if (baseItem instanceof NumberItem) {
+                if (((NumberItem) baseItem).getUnit() != null) {
```
Comment (lsiepel) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java:614]:
```suggestion
            if (baseItem instanceof NumberItem baseAsNumberItem) {
                if (baseAsNumberItem.getUnit() != null) {
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Thanks, minor comment, otherwise LGTM

**Comment by ccutrer** [comment]
> thank you. probably at same point we should remove binding specific setting for temperature unit and rely on item or system settings 
> 
That's the goal! I think we wait one more full cycle though

**Review by lsiepel — APPROVED** [review]
LGTM

**Pull request merged**